scripts/allowlist/README.md

```diff
@@ -0,0 +1,79 @@
+# Genesis File Content
+
+This directory contains a script (`createContent.js`) that generates some content that needs to be added to the genesis 
+file of a network that makes use of the allowlist smart contracts (`contracts/allowlist`).
+
+### Install the dependencies
+
+To install the dependencies of the script run `npm install` in the `scripts/allowlist` directory. 
+
+### Input File
+
+The script reads the file `allowedAccountsAndValidators.txt` that defines in each line the address of an allowed account and
+(optionally) the validator for that account. The account and the validator are specified using their address as a 
+hexadecimal string. If the account has a nominated validator, the account and the validator hexadecimal strings need
+to be separated by a comma:
+
+    0x5B38Da6a701c568545dCfcB03FcB875f56beddC4, 0xAb8483F64d9C6d1EcF9b849Ae677dD3315835cb2
+    0xCA35b7d915458EF540aDe6068dFe2F44E8fa733c
+    0x4b20993Bc481177ec7E8f571ceCaE8A9e22C02db, 0x79731D3Ca6b7E34aC0F824c42a7cC18A495cabaB
+
+The two lines above specify three accounts on the allow list, as well as the validators for the first and third account.
+
+### Run the Script
+
+To run the script in the `scripts/allowlist` directory: `node createContent.js`
+###Output
+
+The script creates a file named `Storage.txt`. The content of this file for the example above will look like this:
+
+    "0x<Address of Contract>": {
+      "balance": "0x00",
+      "code": "0x<Contract Code>"
+      "storage": {
+        "0000000000000000000000000000000000000000000000000000000000000000": "0000000000000000000000000000000000000000000000000000000000000002",
+        "290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563": "000000000000000000000000ab8483f64d9c6d1ecf9b849ae677dd3315835cb2",
+        "290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e564": "00000000000000000000000079731d3ca6b7e34ac0f824c42a7cc18a495cabab",
+        "36306db541fd1551fd93a60031e8a8c89d69ddef41d6249f5fdc265dbc8fffa2": "0000000000000000000000000000000000000000000000000000000000000101",
+        "58d9a93947083dcdedec58d43912ce0326f251a85b7701c5de5bc7d7a150676e": "0000000000000000000000000000000000000000000000000000000000000001",
+        "e20f19dc6931eb9e42fe3f21abe1a9ef59942d8e586871d88564d0d0b63a5e5c": "0000000000000000000000000000000000000000000000000000000000010101",
+        "0000000000000000000000000000000000000000000000000000000000000002": "0000000000000000000000000000000000000000000000000000000000000003"
+      },
+      "version": "0x01"
+    }
+
+The content of the file needs to be placed in the genesis file for the network. In addition the <_Address of Contract_> 
+and <_Contract Code_> need to be filled in.
+
+An example of a genesis file using QBFT can be found in the `genesis.json` file in this directory.
+
+* The <_Address of Contract_> needs to be the same that is stated in the _qbft_ section of the genesis file for the _validatorcontractaddress_.
```

Comment:
This explanation is not too easy to understand. Might it make more sense to instead generate an entire valid genesis file instead of giving the user a natural language algorithm to manually do so?

I would expected the user to copy parts of this genesis file into their instead of directly using it. But having an entire genesis file makes it obvious where certain data belongs. Having it programmatically generated makes it 100% clear how it is derived--*the code is the spec* in a good way.

Reply:
There is a genesis.json file!